diff --git i/include/ollama.hpp w/include/ollama.hpp
index eaa7a51..0f343b8 100644
--- i/include/ollama.hpp
+++ w/include/ollama.hpp
@@ -347,10 +347,7 @@ namespace ollama
                 return type;
             }
 
-            //operator std::string() const { return this->as_simple_string(); }
-            operator std::__cxx11::basic_string<char>() const { return this->as_simple_string(); }
-            //const operator std::string() const { return this->as_simple_string(); }           
-
+            operator std::string() const { return this->as_simple_string(); }
 
         private:
 
